### PR TITLE
Robofac Hybrid Weapons Platform

### PR DESCRIPTION
**Pull request by bombasticSlacks**
#### Summary
Content "Robofac Personnel Use a Custom Weapons Platform"

#### Purpose of change
It always bummed me out that Hub 01 operators just use kitted Military AR's considering the tech and know how they are supposed to have. I know bellow ground one of them uses a rivtech weapon but those don't make sense as a long term solution for kitting operatives as caseless ammo isn't readily available. As a faction with some insight into how the world is going and how bad it is going to get I think it would make sense for them to when performing field operations use a weapon that feels both very modern but also is very practical in the field while still seeming like an over engineered/expensive prototype that contractors and nerds would come up with while having no oversight.

#### Describe the solution
Introducing the Hub 01 Hybrid Weapons Platform or Hub 01 HWP.

The HWP is a modern polymer/metal weapon platform loosely inspired by the FN2000 in ergonomics/sci-fi design
image

 with the unique feature of being quickly modifiable in the field to take either:

- 5.56 Nato
- 7.62 Nato
- 9mm parabellum
- 20x70mm shotshells

All of which would be identified as common rounds able to find and scavenge while in the field. Allowing operatives to act autonomously longer while maintaining full functionality.

Within just a few minutes the barrel, part of the receiver and the magazine well adapter can be replaced allowing for firing of a completely separate cartridge, assuming you also configure the recoil systems settings for appropriate gas levels. This is done with a proprietary barrel swap kits. These can be stored in a simple tool kit designed to hold 3 separate barrels + enclosed receiver bits + magazine well adapters for each caliber not currently installed in the gun. Since the gun is a bullpup **This tool kit is quite long **, it would have to, to hold those barrels and parts. 

The gun does require separate magazines for each type of round. The 4 magazines are:
- HWP 40-round 5.56 magazine: a long mag inspired by 40 round stanags (but not interchangeable)
- HWP 60-round 9mm helical magazine: a compact helical drum inspired by the 50 round mp5 drum mags (but not interchangeable)
- HWP 10-round shotgun magazine: a thick single stack magazine inspired by saiga mags (but not interchangeable)
- HWP 20-round .308 magazine: a double stack heavy AR mag inspired by the FAL and Heavy Scar (but not interchangeable)

It also has a few built in mods. This simplifies the crafting process but also makes sense for an all in one system. It has:
- An integrated forward grip (probably something over designed like the HERA Arms CQR 
image
- Integrated laser sight. I imagine this would be built into the frame of the gun and batteries would be accessible in the grip or something.

It also has a removable Military Hybride 4x/1x. Which is an acog that accepts a top mounted holosight.  I imagine it would be a HAMR since that's a bit more sci-fi than the TRIJICON ones. 
image

Also all of the barrels have a build in muzzle device. 
Assault : Muzzle Break
DMR: Flash Hider
CQC: Choke
PDW: Silencer

**all of these mods are aside from name, description and being non-removable direct copies of other mods from the game.**

This can be obtained by noticing it on the Hub Merc, and then either getting to the end of the Hub quest line where you get access to prototyping, or doing the Merc quest and getting him to vouch for you.

Then you can buy it for 8 coins and 2 of each mag. This takes 7 days. 

TODO:

- [x] Get @John-Candlebury to give this the 👍 
- [x] Put this weapon on the Hub mercenary
- [x] Make it so when talking to the mercenary you can notice his sweet custom gun
- [x] Make it so you can then talk to the intercom about said gun
- [x] Make getting your own copy of the platform a project you can order from the intercom that will take (I'm thinking a week)
- [x] Generic Guns Migrations

#### Describe alternatives you've considered
Making one weapon that's just a tuned 5.56 gun or laser weapon but I think this is a more creative solution for Hub 01 that doesn't introduce any new super guns but still feels futuristic.

Lots of tuning possible on mag counts, lengths (+ other numbers), mods, etc.

Leaving this alone.

~~Implementing this another way than crafting. However I think crafting is the most robust approach that clearly makes you have all the pieces.~~

I swapped to the mod system after fixing a bunch of stuff

#### Testing
Currently is WIP while I add the prototype as a thing that can be procured in game.

#### Additional context
Screen shots of items in game:
Base Rifle no barrel. It shouldn't have gun stats but I would need to do a lot of testing to make sure setting that up wouldn't break any tools in game. **It won't take mags tho** in this state.
image

ASSAULT
image

PDW
image

DMR
image

CQC
image

Toolkit:
image

Loads of dialogue:
image

### CODE REVIEW

File: data/json/items/gun/robofac_gun.json [hunk 1]
```diff
@@ -0,0 +1,178 @@
+[
+  {
+    "abstract": "robofac_gun",
+    "copy-from": "rifle_semi",
+    "looks_like": "ar15",
+    "type": "GUN",
+    "name": { "str": "Hub 01 HWP" },
+    "description": "The Hub 01 Hybrid Weapons Platform.  Made with a combination of dark finished steel, blue polymer furniture, and a number of proprietary bolts and fixings to hold the pieces in place this weapon looks like it could come from a science fiction movie.  This version has no installed barrel and is clearly missing a part on the bullpup magazine well.",
```
Comment (chaohedgehogs) [inline data/json/items/gun/robofac_gun.json:8]:
```suggestion
    "description": "The Hub 01 Hybrid Weapons Platform.  Made with a combination of dark finished steel, blue polymer furniture, and a number of proprietary bolts and fixings to hold the pieces in place, this weapon looks like it belongs in a science fiction movie.  This version has no installed barrel and is clearly missing a part on the bullpup magazine well.",
```

File: data/json/items/gun/robofac_gun.json [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    "name": { "str_sp": "Hub 01 HWP assault" },
+    "description": "The Hub 01 Hybrid Weapons Platform.  Made with a combination of dark finished steel, blue polymer furniture, and a number of proprietary bolts and fixings to hold the pieces in place this weapon looks like it could come from a science fiction movie.  This version has a long barrel with 5.56 ASSAULT stamped into the metal."
+  },
+  {
+    "id": "robofac_gun_smg",
+    "copy-from": "robofac_gun",
+    "looks_like": "ar15",
+    "type": "GUN",
+    "name": { "str_sp": "Hub 01 HWP personal defense" },
+    "description": "The Hub 01 Hybrid Weapons Platform.  Made with a combination of dark finished steel, blue polymer furniture, and a number of proprietary bolts and fixings to hold the pieces in place this weapon looks like it could come from a science fiction movie.  This version has a short barrel with parabellum PDW stamped into the metal.",
```
Comment (chaohedgehogs) [inline data/json/items/gun/robofac_gun.json:44]:
```suggestion
    "description": "The Hub 01 Hybrid Weapons Platform.  Made with a combination of dark finished steel, blue polymer furniture, and a number of proprietary bolts and fixings to hold the pieces in place, this weapon looks like it belongs in a science fiction movie.  This version has a short barrel with parabellum PDW stamped into the metal.",
```

File: data/json/items/gun/robofac_gun.json [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    "modes": [ [ "DEFAULT", "semi-auto", 1 ], [ "AUTO", "auto", 4 ] ],
+    "pocket_data": [ { "pocket_type": "MAGAZINE_WELL", "item_restriction": [ "robofac60" ] } ]
+  },
+  {
+    "id": "robofac_gun_dmr",
+    "copy-from": "robofac_gun",
+    "looks_like": "ar15",
+    "type": "GUN",
+    "name": { "str_sp": "Hub 01 HWP designated marksman" },
+    "description": "The Hub 01 Hybrid Weapons Platform.  Made with a combination of dark finished steel, blue polymer furniture, and a number of proprietary bolts and fixings to hold the pieces in place this weapon looks like it could come from a science fiction movie.  This version has a longer barrel with 7.62 NATO DMR stamped into the metal.",
```
Comment (chaohedgehogs) [inline data/json/items/gun/robofac_gun.json:57]:
```suggestion
    "description": "The Hub 01 Hybrid Weapons Platform.  Made with a combination of dark finished steel, blue polymer furniture, and a number of proprietary bolts and fixings to hold the pieces in place, this weapon looks like it belongs in a science fiction movie.  This version has a longer barrel with 7.62 NATO DMR stamped into the metal.",
```

File: data/json/items/gun/robofac_gun.json [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    "modes": [ [ "DEFAULT", "semi-auto", 1 ] ],
+    "pocket_data": [ { "pocket_type": "MAGAZINE_WELL", "item_restriction": [ "robofac20" ] } ]
+  },
+  {
+    "id": "robofac_gun_shotgun",
+    "copy-from": "robofac_gun",
+    "looks_like": "ar15",
+    "type": "GUN",
+    "name": { "str_sp": "Hub 01 HWP close quarters" },
+    "description": "The Hub 01 Hybrid Weapons Platform.  Made with a combination of dark finished steel, blue polymer furniture, and a number of proprietary bolts and fixings to hold the pieces in place this weapon looks like it could come from a science fiction movie.  This version has a wide barrel with 00 CQC stamped into the metal.",
```
Comment (chaohedgehogs) [inline data/json/items/gun/robofac_gun.json:70]:
```suggestion
    "description": "The Hub 01 Hybrid Weapons Platform.  Made with a combination of dark finished steel, blue polymer furniture, and a number of proprietary bolts and fixings to hold the pieces in place, this weapon looks like it belongs in a science fiction movie.  This version has a wide barrel with 00 CQC stamped into the metal.",
```

**Comment by John-Candlebury** [comment]
Yeah I do like this very much, you do get my 👍 !
Finally I can throw away all my other guns.

**Comment by andrei8l** [comment]
The tools equivalent to that gun kit weight 1.8kg so 200g is supposed to cover all the gun bits for 4 guns?
`retool_ar15_300blk` weighs 1kg and that should be similar to the bits required to convert this HWP from 5.56mm to 7.62mm.

And I have to wonder, if this kind of field conversion for vastly different cartridges is possible, why are there no RL examples?

**Comment by bombasticSlacks** [comment]
First I can increase the weight. I copied the gunsmith kit but you are right it is low.

Conversions like this are cutting edge and currently slightly speculative. A real world example old enough that it's publicized would be the MCX Spear and it's other versions. 

The requirements are estimated by firearms expert Jonathan Ferguson here https://youtube.com/clip/UgkxVsU0tJ7rBn9AdsFRX5m_D6OUi2hHXt0e

Wildly seperate calibers like this wouldn't ever be practical for soldiers. Since a soldier can just carry 9 magazines of 5.56 or 7.62 or whatever a functional cartridge is. They also can have backup from someone with a dedicated shotgun or DMR because ammo and personnel aren't in short supply. 

The lore origin for this weapon will be it was originally a prototype by a manufacturer working on a unified platform for the US. Similar to how the Kalashnikov family of guns all have similar ergonomics and design. So this "quick swap" would have been a functionality on the test models for rapid iteration. It would have never been meant to see actual production since a soldier carrying around 5kg of barrels could just be carrying more ammo. However that doesn't mean the Hub can't make use

**Comment by ISuckM8** [comment]
Is this going to be locked to Hub-01 quests or are you going to add it to their barter menu?

**Comment by bombasticSlacks** [comment]
I can't see the original question anymore but in case anyone else is wondering how this will be handled in game for obtaining it's:

Once the Hub Merc spawns (after at least 1 quest done maybe more) when you talk to them if you have high perception, marksmanship, or mechanics you'll have a dialogue option like "[notice high end gun]". That will prompt an internal monologue where you're character notices that it's not something they've seen before. After this you can talk to the merc about it and with high social get some info from them. Or you can ask the intercom.

The intercom currently after completing the nanofab quest has a new dialogue tree for "prototyping equipment" where you can have the faction building you something/working on a research project but only have one thing building at any given moment. 

This will be something you get prototyped for X Hub Coins and it'll take X days (I'm thinking 5 Coins 7 days) for the gun, manual, toolkit and 2 of each mag since their isn't any development going on just printing and assembly. With additional 2 mags of any type procurable in 1 day for 1 Coin. With the addition of this I might move the prototyping from completing the Hub 01 quests (that are in currently) to completing 2 of 4 but the ---

--- TLDR is: It will take completing missions, money and time to get this item. 

As an aside you could kill the Hub Merc for his but he won't have the user manual on him so you'll just have a bulky AR with non-stanag compliant mags and a 95cm long bag that functions as a screwdriver.  It's another reason I really like crafting as the solution for this system because there's no way you could get the configuration settings right without a manual.

**Comment by John-Candlebury** [comment]
Progression wise, the Merc has their own short quest where you help them fight some monsters nearby. You could have them repay you by vouching for you to the intecom and then have the intercom offer the gun from that point on.

**Comment by kevingranade** [comment]
I'm totally fine with the concept. :+1:
I would actually suggest making it even more flexible by breaking out the different calibre deep m9fs 8f you haven'tvakready, though that might l8se you the tool requirement so meh

**Comment by bombasticSlacks** [comment]
> Progression wise, the Merc has their own short quest where you help them fight some monsters nearby. You could have them repay you by vouching for you to the intecom and then have the intercom offer the gun from that point on.

thanks I'll go with this as the option. 

with that said I'd also like to at some point make it so that Hub 01 will recognize you as a field operative and give you a 'call sign' so that they stop treating you like a stranger I recently added the ability for dialogue to embed user variables #54572 so you could pick from a few themed call tags and then the intercom would refer to you by them. I haven't brainstormed the whole list but like "Tango", "Dark Days", "Wanderer", "Fixer" stuff like that. 

And then the Intercom would say stuff like: "tango we've got another field opportunity for you"

Maybe the merc vouching might be a good kickoff to getting sort of recognized as a reliable agent?

**Comment by bombasticSlacks** [comment]
> I would actually suggest making it even more flexible by breaking out the different calibre deep m9fs 8f you haven'tvakready, 
@kevingranade I am having a lot of difficulty parsing what you meant with this.

**Comment by ISuckM8** [comment]
> I can't see the original question anymore but in case anyone else is wondering how this will be handled in game for obtaining it's:
> 
> Once the Hub Merc spawns (after at least 1 quest done maybe more) when you talk to them if you have high perception, marksmanship, or mechanics you'll have a dialogue option like "[notice high end gun]". That will prompt an internal monologue where you're character notices that it's not something they've seen before. After this you can talk to the merc about it and with high social get some info from them. Or you can ask the intercom.
> 
> The intercom currently after completing the nanofab quest has a new dialogue tree for "prototyping equipment" where you can have the faction building you something/working on a research project but only have one thing building at any given moment.
> 
> This will be something you get prototyped for X Hub Coins and it'll take X days (I'm thinking 5 Coins 7 days) for the gun, manual, toolkit and 2 of each mag since their isn't any development going on just printing and assembly. With additional 2 mags of any type procurable in 1 day for 1 Coin. With the addition of this I might move the prototyping from completing the 4 Hub 01 intercom quests (that are in currently) to completing 2 of 4 but the ---
> 
> --- TLDR is: It will take completing missions, money and time to get this item.
> 
> As an aside you could kill the Hub Merc for his but he won't have the user manual on him so you'll just have a bulky AR with non-stanag compliant mags and a 95cm long bag that functions as a screwdriver. It's another reason I really like crafting as the solution for this system because there's no way you could get the configuration settings right without a manual.

Asked the original question, accidentally deleted it while doing work. Thanks for the answer, helps clear things up.
